Source/GSStandardWindowDecorationView.m

```diff
@@ -379,23 +374,58 @@ - (BOOL) acceptsFirstMouse: (NSEvent*)theEvent
   return YES;
 }
 
+- (BOOL) pointInContentRect:(NSPoint)point
+{
+    return NSPointInRect(point, contentRect);
+}
+
+- (BOOL) pointInTitleBarRect:(NSPoint)point
+{
+  return !NSPointInRect(point, closeButtonRect) &&
```

Comment:
Why are you excluding the buttons here?

Reply:
the function should return YES when the cursor is inside the titlebar area that if clicked activates the window dragging. In the previous implementation this check is implicit given the view hierarchy, the close button is catching the event before the window decoration view

Reply:
I would prefer if we leave the buttons out here. That way we have the old code for non-Wayland. And even with Wayland it should not change the behaviour as you already pointed out as the view hierarchy will take care of the correct delivery of the mouse event.

Reply:
sounds good, i'll update it

Reply:
it is now updated